components/clarity-repl/src/repl/hooks/logger.rs

```diff
@@ -0,0 +1,55 @@
+use clarity::vm::contexts::{Environment, LocalContext};
+use clarity::vm::errors::Error;
+use clarity::vm::functions::NativeFunctions;
+use clarity::vm::types::Value;
+use clarity::vm::SymbolicExpressionType::List;
+use clarity::vm::{EvalHook, ExecutionResult, SymbolicExpression};
+
+use crate::repl::clarity_values::value_to_string;
+
+#[derive(Default, Clone)]
+pub struct LoggerHook;
+
+impl LoggerHook {
+    pub fn new() -> Self {
+        Self
+    }
+}
+
+impl EvalHook for LoggerHook {
+    fn will_begin_eval(&mut self, _: &mut Environment, _: &LocalContext, _: &SymbolicExpression) {}
+
+    fn did_finish_eval(
+        &mut self,
+        env: &mut Environment,
+        _context: &LocalContext,
+        expr: &SymbolicExpression,
+        res: &Result<Value, Error>,
+    ) {
+        let List(list) = &expr.expr else { return };
+        let Some((function_name, _args)) = list.split_first() else {
+            return;
+        };
+        let Some(function_name) = function_name.match_atom() else {
+            return;
+        };
+
+        if let Some(NativeFunctions::Print) = NativeFunctions::lookup_by_name(function_name) {
+            let contract_name = &env.contract_context.contract_identifier.name;
+            let span = &expr.span;
+            let line_annotation = format!("({}:{})", contract_name, span.start_line);
```

Comment:
Clippy is going to complain about this now

```suggestion
            let line_annotation = format!("({contract_name}:{})", span.start_line);
```